### DISCUSSION TITLE
How to define a temporal PDE library with multiple trajectories?

### DISCUSSION THREAD

**Discussion by itsjacobhere** [dynamicslab/pysindy#323] [Q&A] [answered]
I understand when fitting the data multiple trajectories should be passed as a list each with (num_samples, num_features) such as:

`model.fit(X, t=t, multiple_trajectories=True)`

where both X and t are the list of trajectories. When defining a PDE lib such as:

`pde_lib = ps.PDELibrary(
    derivative_order=2,
    implicit_terms=True,
    temporal_grid=t,
    is_uniform=True,
    include_bias=True
)`

how do I pass the proper temporal grid because each trajectory contains a different number of samples?

**Comment by znicolaou** [comment, marked as answer]
Hi there @itsjacobhere. Enabling differing (spatio)temporal grids for different trajectories is on our list of features to implement (#222), so unfortunately, it isn't possible yet. I've been wanting to do this for a while, and  it isn't that hard, but it will require a lot of small changes that will need to be validated. 

Instead, the best current solution is to interpolate the trajectories onto a common temporal grid. If you have data `x1` and `x2` on grids `t1=np.arange(0, T1, dt1)` and `t2=np.arange(0,T2,dt2)`, then maybe define `t=np.arange(0,np.min(T1,T2),np.min(dt1,dt2))` and use `scipy.interpolate.interp1d` to produce `x1_interp=interp1d(x1,t)` and `x2_interp=interp1d(x2,t)`, then `model.fit([x1_interp,x2_interp],t=t)`.

**Reply by itsjacobhere** [reply]
@znicolaou thanks for the quick reply! Doing something like that was my alternative solution as well, originally I was thinking to just truncate or force each trajectory to be the same size, but I wasn't sure if just passing in one (universal) temporal trajectory into the PDE lib would work and if I still need to pass a list for t the same size as X into the `model.fit`, if that makes sense?

I like your idea of interpolating along one temporal grid though, I'll try that for now. Yeah would be a great feature! thanks for all your work maintaining this package!
